src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ArrayConverter.cs

```diff
@@ -48,7 +48,9 @@ protected override bool OnWriteResume(Utf8JsonWriter writer, TCollection value,
                 // Fast path that avoids validation and extra indirection.
                 for (; index < array.Length; index++)
                 {
-                    elementConverter.Write(writer, array[index], options);
+                    Debug.Assert(array[index] != null);
```

Comment:
For converters that support nullable reference types (which our internal/built-converters do), this won't be true.
```C#
string[] strs = new string[] { "foo", null };
string json = JsonSerializer.Serialize(strs);
```

Reply:
I'll then remove this and add a bang.